### PR TITLE
base page object model for proxying locator calls

### PR DESCRIPTION
**Pull request by fungairino**
## What does this PR do?

This PR introduces a new base class `BasePageObject` for page objects in the end-to-end tests of the PixieBrix extension. The `ExtensionsShortcutsPage` and `FloatingActionButton` classes are refactored to extend this new base class. The base class provides passthrough calls to the `Locator` methods, allowing the derived classes to directly call the methods of `Locator` on `this`. Calls are scoped to the root locator created on the page object models.

## Discussion

The main approach taken in this PR is to use a base class with call through methods to the root locator. This allows the derived classes to directly call the methods of `Locator` on `this`, making the code more concise and easier to read.

## Checklist

- [X] Designate a primary reviewer @mnholtz

### CODE REVIEW

File: end-to-end-tests/pageObjects/extensionsShortcutsPage.ts [hunk 1]
```diff
@@ -57,88 +59,82 @@ export class ExtensionsShortcutsPage {
   }
 
   async goto() {
-    await this.page.goto(this.pageUrl);
+    await this.page().goto(this.pageUrl);
```
Comment (fungairino) [inline end-to-end-tests/pageObjects/extensionsShortcutsPage.ts:62]: I'd prefer if we could avoid needing to use `page()` and leave it as an attribute, `page`. 

Technically page can change or close due to navigations and such I suppose though.

**Comment by fungairino** [comment]
Alternatively we could just use this solution which basically is the same thing I implemented here, but nicer

https://github.com/lariat-js/playwright/tree/main

**Comment by grahamlangford** [comment]
> Alternatively we could just use this solution which basically is the same thing I implemented here, but nicer
> 
> https://github.com/lariat-js/playwright/tree/main

33 weekly downloads with a last publish of 7 months ago. Probably not something we want to depend on

**Comment by grahamlangford** [comment]
Can you explain why we need to proxy locator calls?

**Comment by fungairino** [comment]
> Can you explain why we need to proxy locator calls?

Discussed offline - I'll instead make this simpler and just have a passthrough interface to the main locator methods. We lose out on the potentially nice way to write default expectations around out BasePageObject treating them as locators, but not a huge deal since we can just access the rootLocator and do the same.

**Comment by mnholtz** [comment]
What problem are you trying to solve @fungairino ?

**Comment by fungairino** [comment]
@mnholtz The objective here is to make page objects more consistent, easier to create, extend and nest. Avoid redefining the same locator methods in each POM.

**Review by twschiller — APPROVED** [review]

**Pull request merged**